src/clang-c-frontend/clang_c_convert.h

```diff
@@ -267,6 +267,17 @@ class clang_c_convertert
    */
   virtual bool
   get_vft_binding_expr(const clang::MemberExpr &member, exprt &new_expr);
+
+  /*
+   * Function to check whether a clang::FunctionDec represents a
+   * clang::CXXMethodDecl that is virtual OR overrides another function
+   *
+   * Params:
+   *  - fd: a clang::FunctionDec we are currently dealing with
+   *
+   * For C, it always return false.
+   */
+  virtual bool is_fd_virtual_or_overriding(const clang::FunctionDecl &fd);
```

Comment:
Shouldn't this function be local to the CPP frontend then?

Reply:
The get_function in clang-c-convert needs this function to continue for virtual/overriding method as we need its type. 

I also considered moving everything local tp CPP frontend.... But if this function was local to CPP frontend, we would need to add some duplicate code in `clang_cpp_convertert::get_function` that does the same job as the code in `clang_c_convertert::get_function`. I reckon adding one additional condition in clang_c_converter is better than adding some duplicate code in the  overriding function in clang_cpp_converter.

Reply:
> we would need to add some duplicate code in clang_cpp_convertert::get_function that does the same job as the code in clang_c_convertert::get_function. 

You could refactor the duplicated part into another function that should belong to the C frontend. If get_function from C works differently from CPP then it is expected that they have different specializations.

Reply:
Somehow I missed this comment from @rafaelsamenezes in my email inbox. 

>You could refactor the duplicated part into another function that should belong to the C frontend.

True, it makes sense. This appear to be a better structure. I'll see what I can do when working on this function in future patch.